### ISSUE TITLE
Distributed training, process freeze after initialization

### ISSUE THREAD

**Issue by RweBs** [awslabs/dgl-ke#186]
hello, guys, When I try a sample of distributed training, the process freezes after initialization is complete

```
[root@node01 ~]#  DGLBACKEND=pytorch dglke_dist_train --path ~/my_task --ip_config ~/my_task/ip_config.txt            --num_client_proc 1 --model_name TransE_l1 --dataset FB15k --data_path ~/my_task --hidden_dim 400            --gamma 16.0 --lr 0.01 --batch_size 1000 --neg_sample_size 200 --max_step 500 --log_interval 100            --batch_size_eval 16 --test -adv --regularization_coef 1.00E-07 --num_thread 1


Copy script to remote machine ...
dglke_start_kvserver_kvclient.sh                                                    100%  957     2.4MB/s   00:00
dglke_start_kvserver_kvclient.sh                                                    100%  957     3.0MB/s   00:00
dglke_start_kvserver_kvclient.sh                                                    100%  957     2.4MB/s   00:00
Logs are being recorded at: ckpts/TransE_l1_FB15k_11/train.log
Error: port 30051 is not available.
Error: port 30050 is not available.
Error: port 30051 is not available.
Error: port 30050 is not available.
Error: port 30051 is not available.
Error: port 30050 is not available.
Error: port 30050 is not available.
Error: port 30051 is not available.
Reading train triples....
Finished. Read 116786 train triples.
Partition 0 n_entities: 10579
Partition 0 n_relations: 1345
|Train|: 116786
/usr/local/lib64/python3.6/site-packages/dgl/base.py:25: UserWarning: multigraph will be deprecated.DGL will treat all graphs as multigraph in the future.
  warnings.warn(msg, warn_type)
Total initialize time 10.260 seconds
Logs are being recorded at: ckpts/TransE_l1_FB15k_11/train.log
Reading train triples....
Finished. Read 93084 train triples.
Partition 1 n_entities: 7774
Partition 1 n_relations: 1345
|Train|: 93084
Total initialize time 10.349 seconds
/usr/local/lib64/python3.6/site-packages/dgl/base.py:25: UserWarning: multigraph will be deprecated.DGL will treat all graphs as multigraph in the future.
  warnings.warn(msg, warn_type)
Logs are being recorded at: ckpts/TransE_l1_FB15k_11/train.log
Reading train triples....
Finished. Read 176169 train triples.
Partition 2 n_entities: 11014
Partition 2 n_relations: 1345
|Train|: 176169
Total initialize time 10.661 seconds
/usr/local/lib64/python3.6/site-packages/dgl/base.py:25: UserWarning: multigraph will be deprecated.DGL will treat all graphs as multigraph in the future.
  warnings.warn(msg, warn_type)
Logs are being recorded at: ckpts/TransE_l1_FB15k_11/train.log
Reading train triples....
Finished. Read 97103 train triples.
Partition 3 n_entities: 9597
Partition 3 n_relations: 1345
|Train|: 97103
Total initialize time 10.445 seconds
/usr/local/lib64/python3.6/site-packages/dgl/base.py:25: UserWarning: multigraph will be deprecated.DGL will treat all graphs as multigraph in the future.
  warnings.warn(msg, warn_type)
```
I have waited for more than 20 minutes and there has been no response.

- version:
dgl (0.4.3)
dglke (0.1.0)
torch (1.7.1)

![截屏2021-03-02 下午8 34 12](https://user-images.githubusercontent.com/32388499/109649283-a7797d80-7b96-11eb-9ef4-1da4be25b92e.png)

**Comment by vlad43210** [comment]
I have a similar issue with `dglke_eval`, made a comment here: https://github.com/awslabs/dgl-ke/issues/138

**Comment by classicsong** [comment]
From the log. It seems: Error: port 30051 is not available.

**Comment by vlad43210** [comment]
Sorry, when I run `dglke_eval`, it does not give a message about the ports. Do I need certain ports to be open for `dglke_eval` to run?

**Comment by RweBs** [comment]
Thank you, I found that the manual termination of the training process previously caused port 30050 to be occupied. After manually killing the process that was occupying the port, the training now runs normally !

**Comment by vlad43210** [comment]
Still not working for me :( I checked if I have any occupied ports, I do not. I also restarted the container and just ran the `dglke_eval` command, still getting stuck with no error :(

**Comment by vlad43210** [comment]
OK, I looked over the code, and it looks like it's getting stuck in the multiprocessing version. When I set the number of processes to 1, the evaluation completes successfully. I wonder if this is related to running `dgl-ke` in a container, I have had issues with running other multiprocessing tasks in containers before, although I did just run a basic Python multiprocessing test and it succeeded. Not strictly solved, but I have found a workaround at least :)
